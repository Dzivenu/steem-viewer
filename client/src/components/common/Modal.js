import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Portal from './Portal';
import { COLORS } from '../../constants/styles';

const ModalContainer = styled.div`
  position: fixed; 
  left: 0;
  top: 0;
  width: 100%; 
  height: 100%;
  background-color: rgba(0,0,0,0.4);
  z-index: 1;
  overflow-y: scroll;
`;

const ModalContent = styled.div`
  position: relative;
  top: 0;
  height: auto;
  background-color: ${COLORS.WHITE.WHITE};
  border: 1px solid ${COLORS.WHITE.WHITE_SMOKE};
  border-radius: .5em;
  box-shadow: 6px 9px 42px 0px rgba(0,0,0,0.37);
  margin: 2em auto; 
  width: 800px;
`;

class Modal extends Component {
  static propTypes = {
    hideModal: PropTypes.func.isRequired,
  };

  render() {
    const { hideModal } = this.props;
    return (
      <Portal>
        <ModalContainer onClick={hideModal}>
          <ModalContent>
            {this.props.children}
          </ModalContent>
        </ModalContainer>
      </Portal>
    );
  }
}

export default Modal;
