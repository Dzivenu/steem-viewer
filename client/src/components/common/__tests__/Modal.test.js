import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Modal from '../Modal';

Enzyme.configure({ adapter: new Adapter() });

describe('(Component) Modal', () => {
  describe('with default prop values', () => {
    it('renders and matches snapshot', () => {
      const props = {};
      const wrapper = shallow(<Modal {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
