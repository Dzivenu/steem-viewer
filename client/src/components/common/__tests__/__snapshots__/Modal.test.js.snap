// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Modal with default prop values renders and matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Modal />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": <styled.div
        onClick={undefined}
>
        <styled.div />
</styled.div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <styled.div />,
        "onClick": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <styled.div
          onClick={undefined}
>
          <styled.div />
</styled.div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": <styled.div />,
          "onClick": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
